### ISSUE TITLE
Require Server URL to always be passed to avoid confusion?

### ISSUE THREAD

**Issue by tasn** [svix/svix-webhooks#300]
This will make it less likely for people to get this wrong, though the experience will be slightly worse. Need to think about it.

**Comment by svix-frank** [comment]
> This will make it less likely for people to get this wrong, though the experience will be slightly worse. Need to think about it.

I would say if we do follow this path to instead pass the `Region` and we can just have an enumeration `EU`, `US`, etc,.

**Comment by tasn** [comment]
Yeah, but what if they self-host?

**Comment by svix-frank** [comment]
> Yeah, but what if they self-host?

Yeah I'd say still allow the `serverUrl` optional parameter (which takes precedence), but require the region in the main constructor.

Another option would be have a `Region.Custom` that then allows you to pass the serverURL and we can error if serverUrl is set without `Region.Custom` or vice versa.

**Comment by tasn** [comment]
We ended up doing region auto-detection and improved error messages.
